functions/billing/main.py

```diff
@@ -42,26 +45,42 @@
 # [END functions_billing_limit]
 
 # [START functions_billing_slack]
-
 # See https://api.slack.com/docs/token-types#bot for more info
 BOT_ACCESS_TOKEN = 'xxxx-111111111111-abcdefghidklmnopq'
-
-CHANNEL_ID = 'C0XXXXXX'
+CHANNEL = 'C0XXXXXX'
 
 slack_client = slack.WebClient(token=BOT_ACCESS_TOKEN)
 
 
 def notify_slack(data, context):
     pubsub_message = data
 
-    notification_attrs = json.dumps(pubsub_message['attributes'])
-    notification_data = base64.b64decode(data['data']).decode('utf-8')
-    budget_notification_text = f'{notification_attrs}, {notification_data}'
+    # For more information, see
+    # https://cloud.google.com/billing/docs/how-to/budgets-programmatic-notifications#notification_format
+    try:
+        notification_attr = json.dumps(pubsub_message['attributes'])
+    except KeyError:
+        notification_attr = "No attributes passed in"
 
-    slack_client.api_call(
-      'chat.postMessage',
-      channel=CHANNEL_ID,
-      text=budget_notification_text)
+    try:
+        notification_data = base64.b64decode(data['data']).decode('utf-8')
+    except KeyError:
+        notification_data = "No data passed in"
+
+    # This is just a quick dump of the budget data (or an empty string)
+    # You can modify and format the message to meet your needs
+    budget_notification_text = f'{notification_attr}, {notification_data}'
+
+    try:
+        slack_client.api_call(
```

Comment:
Slack documentation presents a slightly different way of using the client here: https://github.com/slackapi/python-slackclient#sending-a-message-to-slack
The .api_call() function failed when passing in a channel parameter during testing.

```suggestion
        slack_client.chat_postMessage(
                'channel': CHANNEL,
                'text'   : budget_notification_text
                )
        
```

Reply:
Yeah I didn't update that part but it seemed to be working fine passing in a channel name (with #) or channel ID?

Reply:
That is, per the documentation here: https://python-slackclient.readthedocs.io/en/latest/basic_usage.html#calling-any-api-methods